File: python3/transpose_matrix.py
"""
space : O(mn)
time  : O(mn)
"""

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        row, col = len(matrix), len(matrix[0])
        ans = []
        
        for x in range(col):
            line = []
            for y in range(row):
                line.append(matrix[y][x]) 
            ans.append(line)
        
        return ans

File: golang/transpose_matrix.go
package main

func transpose(matrix [][]int) [][]int {
	row, col := len(matrix), len(matrix[0])
	var ans [][]int

	for x := 0; x < col; x++ {
		line := []int{}
		for y := 0; y < row; y++ {
			line = append(line, matrix[y][x])
		}
		ans = append(ans, line)
	}

	return ans
}
